server/container/core/src/test/java/org/apache/james/core/MailImplTest.java

```diff
@@ -80,31 +87,173 @@ public void testConstr3() throws MessagingException {
 
         helperTestInitialState(mail);
         helperTestMessageSize(mail, 0);
-        assertEquals("initial message", mimeMessage.getMessageID(), mail.getMessage().getMessageID());
-        assertEquals("sender", sender, mail.getSender().toString());
-        assertEquals("name", name, mail.getName());
+        assertThat(mail.getMessage().getMessageID()).isEqualTo(mimeMessage.getMessageID());
+        assertThat(mail.getSender().asString()).isEqualTo(sender);
+        assertThat(mail.getName()).isEqualTo(name);
         mail.dispose();
     }
 
     @Test
-    public void testDuplicate() throws MessagingException {
+    public void duplicateShouldCreateANewJavaObject() throws MessagingException {
         MailImpl mail = new MailImpl();
         MailImpl duplicate = (MailImpl) mail.duplicate();
-        assertNotSame("is real duplicate", mail, duplicate);
+
+        assertThat(mail).as("duplicate method should return different objects").isNotSameAs(duplicate);
+
         helperTestInitialState(duplicate);
         helperTestMessageSize(duplicate, 0);
     }
 
     @Test
-    public void testDuplicateNewName() throws MessagingException {
+    public void duplicateShouldNotChangeOriginalMailName() throws MessagingException {
         String newName = "aNewName";
-
         MailImpl mail = new MailImpl();
-        assertFalse("before + after names differ", newName.equals(mail.getName()));
+        assertThat(mail.getName()).as("before + after names should differ").isNotEqualTo(newName);
 
-        MailImpl duplicate = (MailImpl) mail.duplicate(newName);
-        assertEquals("new name set", newName, duplicate.getName());
-        helperTestInitialState(duplicate);
-        helperTestMessageSize(duplicate, 0);
+        mail.duplicate(newName);
```

Comment:
That's the way this test was initially written. 

As noted by Luc it do not test much things. The assertion seems useless, so I removed it.